### PR TITLE
Feature: File Picker Widget

### LINKED ISSUE #1271: Feature: File Picker Widget
### Description
- Create a new widget in the widget ecosystem -  File Picker.
- User should be able to pick a file(s) or use drag and drop 

Properties: 
```
height
width
enableDropzone - bool
enablePicker - bool (input picker)
exposeVariable  : { name, content,  type }
maxFileCount - 2 (e.g) // ! hide and show max file count when enableMultiple is true/false
enableMultiple - bool
fileType - [ '.png', '.json'] (types of file that it could accept)
maxSizeLimit (Bytes)
minSizeLimit (Bytes)
visibility
disabled 
event - 'fileSelected' (fires an event "fileSelected" )

```

### PR DESCRIPTION
**Pull request by arpitnath**
Fixes #1271 

### App:

### CODE REVIEW

**Comment by arpitnath** [comment]
update: https://www.loom.com/share/906550dac4b0468585ef3d3af7e8a164

File: frontend/src/Editor/Components/FilePicker.jsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
+import React, { useEffect, useMemo } from 'react';
+import { useDropzone } from 'react-dropzone';
+import { resolveWidgetFieldValue } from '@/_helpers/utils';
+import { toast } from 'react-toastify';
+
+export const FilePicker = ({ width, height, component, currentState, onComponentOptionChanged, onEvent, darkMode }) => {
+  //* property definations
```
Comment (akshaysasidrn) [inline frontend/src/Editor/Components/FilePicker.jsx:7]: nitpick: definitions*

**Review by akshaysasidrn — APPROVED** [review]
@arpitnath Looks good. Need to decide if and how we should handle case wherein user wouldn't want to set a predefined type.

File: frontend/src/Editor/Components/FilePicker.jsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+  const enableDropzone = component.definition.properties.enableDropzone?.value ?? true;
+  const enablePicker = component.definition.properties?.enablePicker?.value ?? true;
```
Comment (akshaysasidrn) [inline frontend/src/Editor/Components/FilePicker.jsx:9]: we can also expect properties here as well?

**Review by Navaneeth-pk — APPROVED** [review]

**Comment by akshaysasidrn** [comment]
> @arpitnath Looks good. Need to decide if and how we should handle case wherein user wouldn't want to set a predefined type.

On further discussion, found this is already handled if we put empty values for accept fields. 👍🏾

**Comment by arpitnath** [comment]
Will add the details in the docs with a follow up PR

**Pull request merged**